src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardSyncTaskManager.java

```diff
@@ -106,24 +112,46 @@
                 }
             }
 
-            currentTask =
-                    new MetricsCollectingTaskDecorator(new ShardSyncTask(kinesisProxy,
-                            leaseManager,
-                            initialPositionInStream,
-                            cleanupLeasesUponShardCompletion,
-                            ignoreUnexpectedChildShards,
-                            shardSyncIdleTimeMillis,
+            currentTask = new MetricsCollectingTaskDecorator(
+                    new ShardSyncTask(kinesisProxy, leaseManager, initialPositionInStream,
+                            cleanupLeasesUponShardCompletion, ignoreUnexpectedChildShards, shardSyncIdleTimeMillis,
                             shardSyncer, latestShards), metricsFactory);
-            future = executorService.submit(currentTask);
+            future = CompletableFuture.supplyAsync(() -> currentTask.call(), executorService)
+                    .whenComplete((taskResult, exception) -> {
+                        if (exception != null) {
```

Comment:
Minor: it might be helpful to extract the logic in whenComplete to a separate method to make it more readable.

Reply:
Refactored into a separate method.